apis/externalsecrets/v1beta1/secretstore_aws_types.go

```diff
@@ -54,14 +54,24 @@ type AWSJWTAuth struct {
 type AWSServiceType string
 
 const (
-	// AWSServiceSecretsManager is the AWS SecretsManager.
+	// AWSServiceSecretsManager is the AWS SecretsManager service.
 	// see: https://docs.aws.amazon.com/secretsmanager/latest/userguide/intro.html
 	AWSServiceSecretsManager AWSServiceType = "SecretsManager"
-	// AWSServiceParameterStore is the AWS SystemsManager ParameterStore.
+	// AWSServiceParameterStore is the AWS SystemsManager ParameterStore service.
 	// see: https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html
 	AWSServiceParameterStore AWSServiceType = "ParameterStore"
 )
 
+// SecretsManager defines how the provider behaves when interacting with AWS
+// SecretsManager. Some of these settings are only applicable to controlling how
+// secrets are deleted, and hence only apply to PushSecret.
+type SecretsManager struct {
+	// +optional
+	ForceDeleteWithoutRecovery bool `json:"forceDeleteWithoutRecovery,omitempty"`
+	// +optional
+	RecoveryWindowInDays int64 `json:"recoveryWindowInDays,omitempty"`
```

Comment:
Can you please add a comment for these options (what they do, that they apply only to PushSecrets, valid ranges for recovery window, also consider pointing to official AWS docs), that'll end up in the openAPI spec and will help users to explore these options e.g. with `kubectl explain` or other means.

(you have that already below in the example `SecretStore` in the documentation)

Reply:
Done!

I ended up grabbing the descriptions from the AWS API docs. `ForceDeleteWithoutRecovery` is extremely long, so I just added the first paragraph (and linked the docs for both fields).